index.js

```diff
@@ -5,7 +5,7 @@
  * @license  MIT
  */
 
-module.exports = function isBuffer (obj) {
-  return obj != null && obj.constructor != null &&
-    typeof obj.constructor.isBuffer === 'function' && obj.constructor.isBuffer(obj)
+/** @returns {boolean} true if obj is a Buffer */
```

Comment:
a related potential improvement

```suggestion
/**
  * Determine if an object is a Buffer
  * @param {unknown} obj
  * @returns {boolean} true if obj is a Buffer
  */
```

could be used with https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html
along side a tsconfig like

```json
{
  "include": ["*.js"],
  "compilerOptions": {
    "target": "ES2020",
    "lib": ["ES2020"],
    "module": "ES2020",
    "moduleResolution": "node",
    "allowJs": true,
    "checkJs": true,
    "declaration": true,
    "emitDeclarationOnly": true
  }
}
```

to automatically generate the index.d.ts file, rather than having it hand-coded